src/config.ts

```diff
@@ -225,26 +227,44 @@ export function isAfterEpoch(): boolean {
 /**
  * Return the parsed global Github configuration
  */
-export function getGlobalGithubConfig(): GithubGlobalConfig {
+let _globalGithubConfigCache: GithubGlobalConfig | null;
+export async function getGlobalGithubConfig(
+  clearCache = false
+): Promise<GithubGlobalConfig> {
+  if (!clearCache && _globalGithubConfigCache !== undefined) {
+    if (_globalGithubConfigCache === null) {
+      throw new ConfigurationError(
+        'GitHub configuration not found in the config file and cannot be determined from Git'
+      );
+    }
+
+    return _globalGithubConfigCache;
+  }
+
   // We extract global Github configuration (owner/repo) from top-level
   // configuration
-  const repoGithubConfig = getConfiguration().github || {};
+  let repoGithubConfig = getConfiguration(clearCache).github || null;
 
   if (!repoGithubConfig) {
-    throw new ConfigurationError(
-      'GitHub configuration not found in the config file'
-    );
-  }
-
-  if (!repoGithubConfig.owner) {
-    throw new ConfigurationError('GitHub target: owner not found');
+    const configDir = getConfigFileDir();
+    const remotes = await simpleGit(configDir).getRemotes(true);
+    const defaultRemote =
+      remotes.find(remote => remote.name === 'origin') || remotes[0];
+    const remoteUrl = defaultRemote
+      ? GitUrlParse(defaultRemote.refs.push || defaultRemote.refs.fetch)
+      : { source: null };
+
+    if (remoteUrl.source === 'github.com') {
+      repoGithubConfig = {
+        owner: remoteUrl.owner,
+        repo: remoteUrl.name,
+      };
+    }
   }
 
-  if (!repoGithubConfig.repo) {
-    throw new ConfigurationError('GitHub target: repo not found');
-  }
+  _globalGithubConfigCache = Object.freeze(repoGithubConfig);
 
-  return repoGithubConfig;
+  return getGlobalGithubConfig();
```

Comment:
No `await` here because it is recursive. 👍

Reply:
![smart, right?](https://media.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif)